src/mds/Server.cc

```diff
@@ -3506,10 +3506,14 @@ CInode* Server::prepare_new_inode(MDRequestRef& mdr, CDir *dir, inodeno_t useino
       _inode->mode |= S_ISGID;
     }
   } else {
-    _inode->gid = mdr->client_request->get_caller_gid();
+    _inode->gid = (mdr->client_request->get_owner_gid() != (unsigned)-1) ?
+		   mdr->client_request->get_owner_gid() :
+		   mdr->client_request->get_caller_gid();
   }
 
-  _inode->uid = mdr->client_request->get_caller_uid();
+  _inode->uid = (mdr->client_request->get_owner_uid() != (unsigned)-1) ?
+		 mdr->client_request->get_owner_uid() :
+		 mdr->client_request->get_caller_uid();
```

Comment:
Since you have force setting the `owner_{uid/gid}`, so we should `ceph_assert()` instead of setting them to `caller_{uid/gid}` here.

If they are still `-1` there should be buggy in MDS ?

Reply:
I personally can't imagine how `-1` can appear there. I'm happy with any option, including `ceph_assert`.

Are you comfortable with `ceph_assert(_inode->uid != (unsigned)-1)`?

Reply:
In theory this should happen anyway, so this should be okay and could catch bugs.

Reply:
> In theory this should happen anyway, so this should be okay and could catch bugs.

yes, I agree. So, I'm adding `ceph_assert(_inode->uid != (unsigned)-1)`, right? Alternatively, we can
add this check just in `get_owner_uid` implementation.

Reply:
> > In theory this should happen anyway, so this should be okay and could catch bugs.
> 
> yes, I agree. So, I'm adding `ceph_assert(_inode->uid != (unsigned)-1)`, right? Alternatively, we can add this check just in `get_owner_uid` implementation.

It will be more proper to check it here instead.

Reply:
Good. Have fixed.